add ubuntu build job

dogboydog opened the pull request:
I think you'll have to enable workflows from forks to have this run on this PR, but you can check out the results here 
https://github.com/dogboydog/qaseprite/actions

dogboydog committed:
add ubuntu build job

dogboydog committed:
use target qaseprite in cmake

bjorn commented on .github/workflows/ubuntu.yml:
```diff
@@ -0,0 +1,40 @@
+    - name: Build
+      run: |
+        cmake -B build -DCMAKE_BUILD_TYPE=Release
+        cmake --build build --config Release --target qaseprite
```
I guess the `--target qaseprite` isn't necessary since I added `EXCLUDE_FROM_ALL` to the `add_subdirectory(aseprite` call, so we really have only one default target.

bjorn commented on .github/workflows/ubuntu.yml:
```diff
@@ -0,0 +1,40 @@
+        version: 6.8.1
+        arch: linux_gcc_64
+        modules: ''
+        tools: 'tools_qtcreator'
```
We don't need to install Qt Creator, that's only done in the Tiled workflow to get Qbs. Actually I'd prefer to compile against the system Qt anyway, so we can drop this step.

bjorn reviewed (commented):
Thanks for getting that to work! Just some tweaks needed. :-)

Also I think we shouldn't call this workflow "Compile on Ubuntu" because we'll be adding macOS and Windows in the future, so maybe just "Build" would do.

bjorn committed:
Various tweaks

* Renamed to `build.yml` since we expect to add other platforms
* Removed `--target qaseprite` since it is the only default target
* Use system Qt (this uses Qt 5 for now)
* Use system Zlib, Pixman, FreeType and HarfBuzz
* Setup ccache to speed up compile

bjorn committed:
Test install and reduce ccache size

bjorn committed:
Check Ubuntu 22.04 with Qt 5 and Ubuntu 24.04 with Qt 6

Also added 'sudo' to install step.

The testing of the install step also revealed that it is likely not
going to the right place on Ubuntu, since it tries to put it in
/usr/lib/plugins/imageformats.

bjorn committed:
Fix workflow syntax

bjorn committed:
Actually run on different Ubuntu versions...
